<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>570</width>
    <height>535</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBoxShapeRecognising">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>10</y>
      <width>251</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>形状识别</string>
    </property>
    <widget class="QLabel" name="labelPreProcessingMethod">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>预处理方法：</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonCameraIn">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>20</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>摄像头输入</string>
     </property>
    </widget>
    <widget class="QSlider" name="horizontalSliderAreaThreshold">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>110</y>
       <width>141</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>5</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="labelAreaThreshold">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>面积阈值：</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxCameraID">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>80</y>
       <width>141</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="labelCameraID">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>摄像头ID：</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonImSelect">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>选取图片</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBoxPreProcessingMethod">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>50</y>
       <width>161</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>OTSU（默认）</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>自适应二值化</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>CANNY边缘检测</string>
      </property>
     </item>
    </widget>
    <zorder>labelPreProcessingMethod</zorder>
    <zorder>pushButtonCameraIn</zorder>
    <zorder>horizontalSliderAreaThreshold</zorder>
    <zorder>labelAreaThreshold</zorder>
    <zorder>labelCameraID</zorder>
    <zorder>pushButtonImSelect</zorder>
    <zorder>spinBoxCameraID</zorder>
    <zorder>comboBoxPreProcessingMethod</zorder>
   </widget>
   <widget class="QGroupBox" name="groupBoxResetSampleNet">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>160</y>
      <width>251</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>重置神经网络和样本</string>
    </property>
    <widget class="QPushButton" name="pushButtonResetSampleNet">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>231</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>重置</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxNumLayers">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>20</y>
       <width>141</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
    </widget>
    <widget class="QLabel" name="labelNumLayers">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>隐含层层数：</string>
     </property>
    </widget>
    <widget class="QLabel" name="labelHiddenLayerSize">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>层节点数：</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxHiddenLayerSize">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>50</y>
       <width>141</width>
       <height>21</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="value">
      <number>9</number>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBoxRenameShape">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>280</y>
      <width>251</width>
      <height>171</height>
     </rect>
    </property>
    <property name="title">
     <string>点击图形纠错</string>
    </property>
    <widget class="QComboBox" name="comboBoxCategory">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>20</y>
       <width>141</width>
       <height>21</height>
      </rect>
     </property>
     <property name="currentIndex">
      <number>-1</number>
     </property>
     <property name="maxVisibleItems">
      <number>10</number>
     </property>
    </widget>
    <widget class="QLabel" name="labelClassify">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>选择类别：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEditAddCategory">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>50</y>
       <width>91</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="labelAddCategory">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>添加类别：</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonAddClassify">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>50</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>添加</string>
     </property>
    </widget>
    <widget class="QLabel" name="labelWeight">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>增加步长：</string>
     </property>
    </widget>
    <widget class="QLabel" name="labelNoiseStrength">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>添加噪声：</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxWeight">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>80</y>
       <width>141</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>30000</number>
     </property>
     <property name="value">
      <number>30</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxNoiseStrength">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>110</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>20</number>
     </property>
     <property name="singleStep">
      <number>0</number>
     </property>
     <property name="value">
      <number>5</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonStartFineTuning">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>231</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>微调神经网络</string>
     </property>
    </widget>
    <widget class="QLabel" name="labelNoiseUnit">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>110</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>像素</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QTextBrowser" name="textBrowserLog">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>20</y>
      <width>251</width>
      <height>491</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBoxAfterChange">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>460</y>
      <width>251</width>
      <height>51</height>
     </rect>
    </property>
    <property name="title">
     <string>保存或还原神经网络和样本</string>
    </property>
    <widget class="QPushButton" name="pushButtonSaveNetAndSample">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>保存</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonLoadNetAndSample">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>20</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>还原</string>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
